dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -99,155 +152,148 @@ public CultureInfo Culture
-    /// <param name="serviceSelector">The <see cref="IAIServiceSelector"/> used to select between multiple AI services.</param>
-    /// <param name="httpHandlerFactory">The <see cref="IDelegatingHandlerFactory"/> to use when constructing <see cref="HttpClient"/> instances for use in HTTP requests.</param>
-    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    #region Helpers on top of Plugins and Services
+    /// <summary>Gets a service from the <see cref="Services"/> collection.</summary>
+    /// <typeparam name="T">Specifies the type of the service to get.</typeparam>
+    /// <param name="serviceId">An object that specifies the key of the service to get.</param>
+    /// <returns>The found service instance.</returns>
+    /// <exception cref="KernelException">A service of the specified type and name could not be found.</exception>
     /// <remarks>
-    /// The KernelBuilder class provides a fluent API for constructing a <see cref="Kernel"/> instance.
+    /// The behavior of this method is not the same as that of <see cref="IServiceProvider.GetService(Type)"/>
+    /// on the exposed <see cref="Services"/>. Rather, it is opinionated view around it. If a <paramref name="serviceId"/>
+    /// is provided, it will attempt to find a service registered with that key. If no <paramref name="serviceId"/>
+    /// is provided, it will attempt to find any service registered, regardless of whether it was registered with
+    /// with a key. If multiple services meet the criteria, it will return one of those registered, but no guarantee
+    /// on exactly which. For certain services, like <see cref="ILoggerFactory"/>, it will also return a default implementation
+    /// if no key was specified and no service was found. If it's able to find the specified service, that service is returned.
+    /// Otherwise, an exception is thrown.
     /// </remarks>
-    public Kernel(
-        IAIServiceProvider aiServiceProvider,
-        IEnumerable<IKernelPlugin>? plugins = null,
-        IAIServiceSelector? serviceSelector = null,
-        IDelegatingHandlerFactory? httpHandlerFactory = null,
-        ILoggerFactory? loggerFactory = null)
+    public T GetService<T>(string? serviceId = null) where T : class
     {
-        Verify.NotNull(aiServiceProvider);
+        T? service = null;
 
-        this.ServiceProvider = aiServiceProvider;
-        this._plugins = plugins is not null ? new KernelPluginCollection(plugins) : null;
-        this._serviceSelector = serviceSelector;
-        this.HttpHandlerFactory = httpHandlerFactory ?? NullHttpHandlerFactory.Instance;
-        this.LoggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+        if (serviceId is not null)
+        {
+            if (this.Services is IKeyedServiceProvider)
```

Comment:
Should we throw a better exception if a `serviceId` is provided, but the service provider isn't an `IKeyedServiceProvider`? Right now it would just say `"Service of type '{typeof(T)}' and key '{serviceId}' not registered."`

Reply:
I can tweak the message for that case.